### PR TITLE
[omada-controller] - Add chart

### PR DESCRIPTION
**Pull request by charlie-haley**
Signed-off-by: Charlie Haley <27928201+charlie-haley@users.noreply.github.com>

**Description of the change**

Adds a chart for the TP-Link omada controller.

**Possible drawbacks**

The app appears to do weird things with the host header, one limitation at the moment is redirects, I'm trying to find a nice way around this, any suggestions are welcome.

Currently I have to add these annotations to my ingress
```
annotations:
    nginx.ingress.kubernetes.io/auth-tls-pass-certificate-to-upstream: "true"
    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
    nginx.ingress.kubernetes.io/upstream-vhost: "omada.mydomain.xyz:8043"
```

When hitting the domain `omada.mydomain.xyz` it will redirect me to the login page, but it'll have the port appended to it `omada.mydomain.xyz:8043/login`, it works if I manually change the URL in my address bar to `omada.mydomain.xyz/login`

Obviously my above solution doesn't apply to everyone and entirely depends on the ingress controller being used. [Someone encountered the same issue and there's a small discussion on Reddit](https://www.reddit.com/r/selfhosted/comments/krsupy/issues_accessing_tplink_omada_controller_software/)

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [X] Chart is using our [common](https://github.com/k8s-at-home/charts/tree/master/charts/common) library as a dependency.
- [ ] (optional) Variables are documented in the README.md

### CODE REVIEW

File: charts/omada-controller/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+keywords:
+- omada-controller
+home: https://github.com/k8s-at-home/charts/tree/master/charts/omada-controller
+icon: https://omada-controller.org/icon
+sources:
+- https://github.com/omada-controller/omada-controller-docker
+- https://github.com/k8s-at-home/charts/tree/master/charts/omada-controller
+maintainers:
+- name: omada-controller
+  email: omada-controller@omada-controller.com
```
Comment (onedr0p) [inline charts/omada-controller/Chart.yaml:16]: This should be a valid maintainer name on github, preferably yours.

File: charts/omada-controller/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v2
+appVersion: 4.2.11
+description: omada-controller helm package
```
Comment (onedr0p) [inline charts/omada-controller/Chart.yaml:3]: a more clear description would be good

File: charts/omada-controller/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+name: omada-controller
+version: 1.0.0
+kubeVersion: ">=1.16.0"
+keywords:
+- omada-controller
+home: https://github.com/k8s-at-home/charts/tree/master/charts/omada-controller
+icon: https://omada-controller.org/icon
```
Comment (onedr0p) [inline charts/omada-controller/Chart.yaml:10]: make sure there's a icon

File: charts/omada-controller/values.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    port: 8043
+
+ingress:
+  enabled: false
+
+# persistence:
+#   data:
+#     enabled: false
+#     emptyDir: false
+#     mountPath: /data
```
Comment (onedr0p) [inline charts/omada-controller/values.yaml:32]: any reason this is all commented out?

**Review by onedr0p — CHANGES_REQUESTED** [review]

File: charts/omada-controller/values.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+ingress:
+  enabled: false
+
+persistence:
+  data:
+    enabled: true
+    emptyDir: false
+    mountPath: /data
+    accessMode: ReadWriteOnce
```
Comment (bjw-s) [inline charts/omada-controller/values.yaml:33]:
```suggestion
persistence:
  data:
    enabled: false
    emptyDir: false
```

This should be sufficient, and will also help the chart pass CI

**Review by onedr0p — APPROVED** [review]

**Pull request merged**